user-org-service/app/controllers/usermanagement/UserController.java

```diff
@@ -15,27 +17,31 @@
  */
 public class UserController extends BaseController {
 
-  /**
-   * This method is used to create user
-   *
-   * @return Return a promise of created userResponse
-   */
-  public CompletionStage<Result> createUser() {
-    startTrace("createUser");
-    CompletionStage<Result> response = handleRequest(request(),null, UserActorOperations.CREATE_USER.getOperation());
-    endTrace("createUser");
-    return response;
-  }
+    /**
+     * This method is used to create user
+     *
+     * @return Return a promise of created userResponse
+     */
+    public CompletionStage<Result> createUser() {
+        startTrace("createUser");
+        CompletionStage<Result> response = handleRequest(request(), (request) -> {
+            JSONObject jsonObject = jsonifyRequestObject(request().body().asJson());
+            new UserModelValidator().validate(jsonObject, request().uri());
+            return null;
+        }, UserActorOperations.CREATE_USER.getOperation());
+        endTrace("createUser");
+        return response;
+    }
 
-  /**
-   * This method is used to create user in this method userName is not mandatory
-   *
-   * @return Return a promise of created userResponse
-   */
-  public CompletionStage<Result> createUserV2() {
-    startTrace("createUserV2");
-    CompletionStage<Result> response = handleRequest(request(),null,UserActorOperations.CREATE_USER.getOperation());
-    endTrace("createUserV2");
-    return response;
-  }
+    /**
+     * This method is used to create user in this method userName is not mandatory
+     *
+     * @return Return a promise of created userResponse
+     */
+    public CompletionStage<Result> createUserV2() {
```

Comment:
As there is no difference between v1 and v2 apart from username mandatory , i think we will do follwoing :
1. in routes will have two endpoint v1 and v2 but both will poin to same controller method
2. and in code will make user name optional , if user won't pass then system will generte it.

Reply:
ya this can be done _

Reply:
but what will happen to the adapters??

Reply:
From controller both v1 and v2 endpoint will call createUser method only.